README.md

```diff
@@ -27,11 +28,208 @@ For each tasks there is a corresponding CmdLet:
+17:
+18: VOLUME /configs
+19:
+20: ENTRYPOINT [ "/usr/local/bin/container-structure-test" ]
+```
+
+This Dockerfile in particular has no linting remarks, so it is just output in its entirety with line numbers. Being a mere human, I will forget things from time to time. Let's imagine I omitted the instruction in line 3 on how to deal with commands that fail in a piped execution and run the same commands:
+
+```powershell
+PS C:\docker> $result = Invoke-DockerLint .\Dockerfile
+PS C:\docker> $result.LintOutput
+1: FROM ubuntu:18.04
+2:
+3: RUN apt-get update \
+4:     && apt-get install -y --no-install-recommends curl=7.* ca-certificates=* \
+5:     && apt-get clean \
+6:     && rm -rf /var/lib/apt/lists/*
+7:
+8: RUN update-ca-certificates
+9:
+DL4006 Set the SHELL option -o pipefail before RUN with a pipe in it
+10: RUN curl -sL https://get.docker.com/ | sh
+11:
+12: RUN curl -LO https://storage.googleapis.com/container-structure-test/v1.8.0/container-structure-test-linux-amd64 \
+13:     && chmod +x container-structure-test-linux-amd64 \
+14:     && mv container-structure-test-linux-amd64 /usr/local/bin/container-structure-test
+15:
+16: VOLUME /configs
+17:
+18: ENTRYPOINT [ "/usr/local/bin/container-structure-test" ]
+```
+
+Now, the linter is no longer happy and it has added a remark just above line 10 instructing us on how to fix the problem. The first part of the message is a unique lint rule id that can be used to find the lint rationale and in-depth explanation on https://github.com/hadolint/hadolint.
+
+This concludes the examples on linting. Whilst linting can help you improve parts of your Docker-style and quality of the images, it is no substitute for real testing.
+
+### Testing a docker image
+We provide testing of docker images using Google's Container Structure framework (https://github.com/GoogleContainerTools/container-structure-test).
+
+To run tests, you first define them in .yml configs. Then you build the image the image you want to test, and then finally you run the tests.
+
+So let's start by building an image called `structure`
+
+```powershell
+PS C:\docker> Invoke-DockerBuild . -ImageName structure
+
+Dockerfile    : Dockerfile
+ImageName     : structure
+Registry      :
+Tag           : latest
+CommandResult : CommandResult
+```
+
+Then, we define a test to check if the correct binary was produced and placed in accordance with the documentation at https://github.com/GoogleContainerTools/container-structure-test.
+
+The config file gcs-commands.yml looks like this:
+
+```yml
+schemaVersion: "2.0.0"
+
+commandTests:
+  - name: "say hello world"
+    command: "bash"
+    args:
+      - -c
+      - |
+        echo hello &&
+        echo world
+    exitCode: 0
+    expectedOutput: ["hello", "world"]
+```
+
+And you run the tests like this:
+
+```powershell
+PS C:\docker> Invoke-DockerTests -ImageName 3shape/containerized-structure-test -ConfigFiles gcs-commands.yml
```

Comment:
Here the value for the parameter `-ImageName`, signals the need of #51 :smile:

Reply:
I suggest this PR is now blocked by #51 🤣